src/stactools/modis/sinusoidal.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import math
+from itertools import groupby
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+from pystac import Item
+from shapely.geometry.polygon import Polygon
+from stactools.core.utils.raster_footprint import RasterFootprint, densify_by_distance
+from stactools.core.utils.round import recursive_round
+
+from stactools.modis.constants import (
+    COLLECTION_FOOTPRINT_METADATA,
+    PRECISION,
+    SINUSOIDAL_SPHERE_RADIUS,
+    SINUSOIDAL_TILE_METERS,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SinusoidalFootprint(RasterFootprint):
+    def densify_polygon(self, polygon: Polygon) -> Polygon:
+        # We densify at pixel distance to ensure that the final footprint
+        # polygon never strays further than the specified `simplify_tolerance`
+        # from the edge of the raster data.
+        return Polygon(densify_by_distance(polygon.exterior.coords, self.transform[0]))
+
+    def reproject_polygon(self, polygon: Polygon) -> Polygon:
+        lonlat_list = sinusoidal_grid_to_lonlat(polygon.exterior.coords)
+        # Conversion from grid to lon/lat for tiles that contain pixels beyond
+        # the projection edge produces coords beyond the antimeridian. Clipping
+        # to the natural extents of the projection cleans this up.
+        lonlat_array = np.asarray(lonlat_list)
+        np.clip(lonlat_array[:, 0], -180, 180, lonlat_array[:, 0])
+        np.clip(lonlat_array[:, 1], -90, 90, lonlat_array[:, 1])
+        polygon = Polygon(
+            recursive_round(lonlat_array.tolist(), precision=self.precision)
+        )
+        polygon = Polygon([k for k, _ in groupby(polygon.exterior.coords)])
+        return polygon
+
+
+def sinusoidal_grid_to_lonlat(
+    grid_coords: List[Tuple[float, float]]
+) -> List[Tuple[float, float]]:
+    """Transform MODIS and VIIRS sinusoidal projection grid coordinates to
+    spherical longitude and latitude.
+
+    Args:
+        grid_coords (List[Tuple[float, float]]): List of sinusoidal projection grid
+            coordinate tuples in (x, y) order.
+
+    Returns:
+        List[Tuple[float, float]]: List of spherical longitude and latitude
+            coordinate tuples in (longitude, latitude) order.
+    """
+
+    def _sigfigs(x: float, n: int) -> float:
+        # https://discuss.python.org/t/rounding-to-significant-figures-feature-request-for-math-library/16395/82  # noqa
+        return float(f"{x:.{n}g}")
+
+    lonlat = []
+    for x, y in grid_coords:
+        latitude = math.degrees(y / SINUSOIDAL_SPHERE_RADIUS)
+        longitude = math.degrees(
+            x / (SINUSOIDAL_SPHERE_RADIUS * math.cos(y / SINUSOIDAL_SPHERE_RADIUS))
+        )
+        lonlat.append((_sigfigs(longitude, 11), _sigfigs(latitude, 11)))
```

Comment:
Why is 11 necessary? Seems like a lot?

Reply:
Yeah, that was necessary several "versions" ago. At one point I needed to chop numbers off but guarantee sig figs (sig figs != precision), but I see that I can just round to the desired precision now. This will simplify the code. Thanks for raising a flag on this.

Reply:
Removed.